Server/mods/deathmatch/logic/luadefs/CLuaDatabaseDefs.cpp

```diff
@@ -676,16 +676,16 @@ int CLuaDatabaseDefs::ExecuteSQLQuery(lua_State* luaVM)
         CPerfStatSqliteTiming::GetSingleton()->SetCurrentResource(luaVM);
         if (CStaticFunctionDefinitions::ExecuteSQLQuery(strQuery, &Args, &Result))
         {
-            lua_newtable(luaVM);
-            int i = 0;
-            for (CRegistryResultIterator iter = Result->begin(); iter != Result->end(); ++iter, ++i)
+            lua_createtable(luaVM, Result->size(), 0);
+            lua_Number i = 1;
+            for (CRegistryResultIterator iter = Result->begin(); iter != Result->end(); ++iter)
             {
                 const CRegistryResultRow& row = *iter;
-                // for ( int i = 0; i < Result.nRows; i++ ) {
-                lua_newtable(luaVM);                     // new table
-                lua_pushnumber(luaVM, i + 1);            // row index number (starting at 1, not 0)
-                lua_pushvalue(luaVM, -2);                // value
-                lua_settable(luaVM, -4);                 // refer to the top level table
+
+                lua_createtable(luaVM, Result->nColumns, 0);  // new table
```

Comment:
I am worried that this will most frequently overallocate for very wide tables with many columns, with values that are null, which may or may not be a common occurence for MTA scripters

Reply:
I personally think the benefits outweigh the drawbacks. I very rarely see tables with nulls used in MTA, if at all.
Most people use wide tables with non-null values.
Also, keep in mind, that these tables are rarely kept. Usually they're processed once with a pair/ipairs loop, and then the GC takes care of it, so even if we do allocate more than needed it wont last long.